source/javascripts/components/unified-editor/StepSelectorDrawer/components/AlgoliaStepList/hooks/useSearchAlgoliaSteps.ts

```diff
@@ -0,0 +1,34 @@
+import { useQuery } from '@tanstack/react-query';
+import StepApi, { AlgoliaStepResponse } from '@/core/api/StepApi';
+import RuntimeUtils from '@/core/utils/RuntimeUtils';
+import useDebouncedFilter from './useDebouncedFilter';
+
+const { stepsClient } = StepApi.getAlgoliaClients();
+
+const useSearchAlgoliaSteps = () => {
+  const { search, categories } = useDebouncedFilter();
+
+  return useQuery({
+    queryKey: ['search-algolia-steps', { search, categories }] as const,
+    queryFn: async () => {
+      const result = await stepsClient.search<AlgoliaStepResponse>(search, {
+        cacheable: true,
+        hitsPerPage: 1000,
+        analytics: RuntimeUtils.isProduction(),
+        clickAnalytics: RuntimeUtils.isProduction(),
+        facetFilters: [
+          'is_latest:true',
+          'is_deprecated:false',
+          categories.map((category) => `step.type_tags:${category}`),
```

Comment:
Doesn't need this to be a spread?
```suggestion
          ...categories.map((category) => `step.type_tags:${category}`),
```